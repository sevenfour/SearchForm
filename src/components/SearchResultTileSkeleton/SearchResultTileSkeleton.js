import React from 'react';
import PropTypes from 'prop-types';

import clsx from 'clsx';

import styles from './SearchResultTileSkeleton.module.css';

const SearchResultTileSkeleton = props => {

  const { className } = props;

  return (
    <div
      aria-label="Search result Tile Skeleton"
      className={clsx(styles.container, className)}
      role="img"
    >
      <div
        className={clsx(styles.aspectRatioBox, styles.imageBox)}
        style={{'--aspect-ratio': '16/9'}}
      />
      <div className={styles.dataBox} />
    </div>
  );
};

SearchResultTileSkeleton.propTypes = {
  className: PropTypes.string
};

export default SearchResultTileSkeleton;
