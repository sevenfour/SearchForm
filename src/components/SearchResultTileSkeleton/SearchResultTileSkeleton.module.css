
.container {
  display: flex;
  flex-direction: column;
  box-shadow: var(--box-shadow);
  padding: 1rem;
}

/* The following aspect-ratio technique/hack came from this post
 * https://css-tricks.com/aspect-ratio-boxes/
 */

.aspectRatioBox {
  position: relative;
}

.aspectRatioBox::before {
  --aspect-ratio-result: calc(var(--aspect-ratio));
  display: block;
  padding-bottom: calc(100% / var(--aspect-ratio-result));
  content: '';
}

.aspectRatioBox > :first-child {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.imageBox {
  margin: -1rem -1rem 1rem;
  border-bottom: 2px solid var(--stone-grey);
}

.dataBox {
  border: 2px solid var(--stone-grey);
  height: 1rem;
}
