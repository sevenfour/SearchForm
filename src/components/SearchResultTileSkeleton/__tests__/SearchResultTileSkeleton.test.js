import React from 'react';
import { shallow } from 'enzyme';

import SearchResultTileSkeleton from '..';

describe('<SearchResultTileSkeleton /> component tests', () => {

  it('renders', () => {
    const skeleton = shallow(
      <SearchResultTileSkeleton />
    );

    expect(skeleton).toMatchSnapshot();
  });

});
