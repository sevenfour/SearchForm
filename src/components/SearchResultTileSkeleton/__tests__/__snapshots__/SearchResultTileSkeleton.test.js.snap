// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultTileSkeleton /> component tests renders 1`] = `
<div
  aria-label="Search result Tile Skeleton"
  className="container"
  role="img"
>
  <div
    className="aspectRatioBox imageBox"
    style={
      Object {
        "--aspect-ratio": "16/9",
      }
    }
  />
  <div
    className="dataBox"
  />
</div>
`;
